lib/active_model/serializer/collection_serializer.rb

```diff
@@ -10,8 +10,14 @@ class CollectionSerializer
       def initialize(resources, options = {})
         @root = options[:root]
```

Comment:
@bf4 @NullVoxPopuli  @remear  I really think that `allocate` is not the right solution here. `json_key` by default use `root` or the `object` (https://github.com/rails-api/active_model_serializers/blob/master/lib/active_model/serializer.rb#L185). Neither of these would be available and might result in a bug if the `json_key` method is not overidden. That might also break for polymorphic collections.

I stay on my position that the prefered solution here would be the ones described at https://github.com/rails-api/active_model_serializers/issues/1536#issuecomment-188434945. Or even subclass the collection serializer and add its own root handling.